role/api.py

```diff
@@ -0,0 +1,50 @@
+"""
+Functions for handling roles
+"""
+
+from __future__ import unicode_literals
+
+from django.contrib.auth.models import Group
+from guardian.shortcuts import assign_perm
+
+from role.permissions import RepoPermission, GroupTypes
+
+
+def roles_init_new_repo(repo):
+    """
+    Create new groups for the repo
+    """
+    administrator_group = Group.objects.get_or_create(
```

Comment:
I don't think groups can be created multiple times because the name must to be unique, but there is no harm to wrap everything in a transaction anyway.